File: backend/modulos/metas/aplicacion/EliminarMetaUseCase.py
from modulos.metas.dominio.MetaRepositoryPort import MetaRepositoryPort


class EliminarMetaUseCase:
    def __init__(self, repositorio: MetaRepositoryPort):
        self.repositorio = repositorio

    def ejecutar(self, meta_id: str) -> bool:
        meta = self.repositorio.obtener_por_id(meta_id)
        if not meta:
            raise ValueError('Meta no encontrada.')
        return self.repositorio.eliminar(meta_id)

File: frontend/src/modules/proyectos/application/EliminarMetaUseCase.ts
import { MetaRepositoryPort } from '../domain/MetaRepositoryPort';

export class EliminarMetaUseCase {
  constructor(private readonly repositorio: MetaRepositoryPort) {}

  ejecutar(proyectoId: string, metaId: string): Promise<void> {
    return this.repositorio.eliminar(proyectoId, metaId);
  }
}
